rectools/models/sasrec.py

```diff
@@ -730,15 +745,67 @@ def training_step(self, batch: torch.Tensor, batch_idx: int) -> torch.Tensor:
 
             # Loss output will have a shape of [batch_size, session_max_len]
             # and will have zeros for every `0` target label
-
+            logits = session_embs @ item_embs.T
             loss = torch.nn.functional.cross_entropy(
                 logits.transpose(1, 2), y, ignore_index=0, reduction="none"
             )  # [batch_size, session_max_len]
             loss = loss * w
             n = (loss > 0).to(loss.dtype)
             loss = torch.sum(loss) / torch.sum(n)
             return loss
-        raise ValueError(f"loss {loss} is not supported")
+        if self.loss == "BCE":
```

Comment:
Too much logic in one method.
For all loss options let's do this logic: 
`if self.loss == "BCE":
    return self._calc_bce_loss(batch)
`

Reply:
fixed